src/Connection/Cluster/RedisCluster.php

```diff
@@ -482,24 +510,42 @@ protected function onAskResponse(CommandInterface $command, $details)
      */
     private function retryCommandOnFailure(CommandInterface $command, $method)
     {
-        $failure = false;
+        $retries = 0 ;
+
+        $retryAfter = $this->minRetryAfter;
 
         RETRY_COMMAND: {
             try {
                 $response = $this->getConnectionByCommand($command)->$method($command);
-            } catch (ConnectionException $exception) {
-                $connection = $exception->getConnection();
-                $connection->disconnect();
 
-                $this->remove($connection);
+                if ($response instanceof \Predis\Response\Error) {
```

Comment:
Let's import this class.

Reply:
done [digital2real/predis@fd385bcd2a369567799d421a7111aacf3cc12f6e ](https://github.com/digital2real/predis/commit/fd385bcd2a369567799d421a7111aacf3cc12f6e)